src/shims/unix/sync.rs

```diff
@@ -355,20 +361,30 @@ fn cond_clock_offset<'tcx>(ecx: &MiriInterpCx<'tcx>) -> u64 {
             .offset(Size::from_bytes(offset), ecx.machine.layouts.i32, ecx)
             .unwrap();
         let id = ecx.read_scalar(&id_field).unwrap().to_i32().unwrap();
+        let id = translate_clock_id(ecx, id).expect("static initializer should be valid");
         assert!(
-            is_cond_clock_realtime(ecx, id),
+            matches!(id, ClockType::Realtime),
             "PTHREAD_COND_INITIALIZER is incompatible with our pthread_cond layout: clock is not CLOCK_REALTIME"
         );
     }
 
     offset
 }
 
+#[derive(Debug, Clone, Copy)]
+enum ClockType {
```

Comment:
We call it the clock "ID" in most places, and the C type name for this is `clockid_t`. So `ClockId` seems like a better name for this type?